modules/asg_node_group/variables.tf

```diff
@@ -17,6 +17,11 @@ variable "name" {
   type        = string
   default     = ""
   description = "An optional identifier for this node group"
+
+  validation {
```

Comment:
Maybe covered by https://github.com/cookpad/terraform-aws-eks/pull/261/files#r750337723 already (and vice versa). This validation is more intuitive to a user

Reply:
The vice versa doesn't apply, as if this isn't present, the label is calculated.

Although this is redundant, it seemed more user friendly to duplicate it here. Also, I think the native tf validation fail is nicer.